hyperspy/_signals/signal2d.py

```diff
@@ -605,3 +608,433 @@ def add_ramp(self, ramp_x, ramp_y, offset=0):
+        binarised_image = np.zeros(image.shape)
+        binarised_image[image > 0.5] = 1
+        return binarised_image
+
+    def separate_peaks(binarised_image):
+        """Identify adjacent 'on' coordinates via DBSCAN."""
+        bi = binarised_image.astype('bool')
+        coordinates = np.indices(bi.data.shape).reshape(2, -1).T[
+            bi.flatten()]
+        db = DBSCAN(2, 3)
+        peaks = []
+        labeled_points = db.fit_predict(coordinates)
+        for peak_label in list(set(labeled_points)):
+            peaks.append(coordinates[labeled_points == peak_label])
+        return peaks
+
+    def _peak_find_once(image):
+        """Smooth, binarise, and find peaks according to main algorithm."""
+        image = smooth(image)
+        image = half_binarise(image)
+        peaks = separate_peaks(image)
+        return image, peaks
+
+    def stat_peak_finder(image):
+        """Find peaks in diffraction image. Algorithm stages in comments."""
+        image = normalize(image)  # 1
+        image = stat_binarise(image)  # 2, 3
+        n_peaks = np.infty  # Initial number of peaks
+        image, peaks = _peak_find_once(image)  # 4-6
+        m_peaks = len(peaks)  # Actual number of peaks
+        while (n_peaks - m_peaks) / n_peaks > 0.05:  # 8
+            n_peaks = m_peaks
+            image, peaks = _peak_find_once(image)
+            m_peaks = len(peaks)
+        peak_centers = np.array(
+            [np.mean(peak, axis=0) for peak in peaks])  # 7
+        return peak_centers
+
+    return clean_peaks(stat_peak_finder(z))
+
+
+def find_peaks_dog(z, min_sigma=1., max_sigma=50, sigma_ratio=1.6,
+                   threshold=0.2, overlap=0.5):
+    """
+    Finds peaks via the difference of Gaussian Matrices method from
+    `scikit-image`.
+
+    Parameters
+    ----------
+    z : ndarray
+        2-d array of intensities
+    min_sigma, max_sigma, sigma_ratio, threshold, overlap :
+        Additional parameters to be passed to the algorithm. See `blob_dog`
+        documentation for details:
+        http://scikit-image.org/docs/dev/api/skimage.feature.html#blob-dog
+
+    Returns
+    -------
+    ndarray
+        (n_peaks, 2)
+        Array of peak coordinates.
+
+    Notes
+    -----
+    While highly effective at finding even very faint peaks, this method is
+        sensitive to fluctuations in intensity near the edges of the image.
+
+    """
+    from skimage.feature import blob_dog
+    z = z / np.max(z)
+    blobs = blob_dog(z, min_sigma=min_sigma, max_sigma=max_sigma,
+                     sigma_ratio=sigma_ratio, threshold=threshold,
+                     overlap=overlap)
+    try:
+        centers = blobs[:, :2]
+    except IndexError:
+        return NO_PEAKS
+    clean_centers = []
+    for center in centers:
+        if len(np.intersect1d(center, (0,) + z.shape + tuple(
+                        c - 1 for c in z.shape))) > 0:
+            continue
```

Comment:
If I understand this correctly, here you check that the found centers are not at the edges of the image + not at the [far edges - 1]. Why is that? Also, why then not check that the centers are at (1,) coordinates (equivalent of (c-1))?